src/txn/txn.c

```diff
@@ -2369,28 +2369,33 @@ __wt_txn_global_shutdown(WT_SESSION_IMPL *session, const char **cfg)
     WT_SESSION_IMPL *s;
     char ts_string[WT_TS_INT_STRING_SIZE];
     const char *ckpt_cfg;
+    bool use_timestamp;
 
     conn = S2C(session);
+    use_timestamp = false;
 
     /*
      * Perform a system-wide checkpoint so that all tables are consistent with each other. All
      * transactions are resolved but ignore timestamps to make sure all data gets to disk. Do this
      * before shutting down all the subsystems. We have shut down all user sessions, but send in
      * true for waiting for internal races.
      */
+    F_SET(conn, WT_CONN_CLOSING_CHECKPOINT);
     WT_TRET(__wt_config_gets(session, cfg, "use_timestamp", &cval));
     ckpt_cfg = "use_timestamp=false";
     if (cval.val != 0) {
         ckpt_cfg = "use_timestamp=true";
-        if (conn->txn_global.has_stable_timestamp)
+        if (conn->txn_global.has_stable_timestamp) {
             F_SET(conn, WT_CONN_CLOSING_TIMESTAMP);
+            use_timestamp = true;
+        }
     }
     if (!F_ISSET(conn, WT_CONN_IN_MEMORY | WT_CONN_READONLY | WT_CONN_PANIC)) {
         /*
          * Perform rollback to stable to ensure that the stable version is written to disk on a
          * clean shutdown.
          */
-        if (F_ISSET(conn, WT_CONN_CLOSING_TIMESTAMP)) {
+        if (use_timestamp) {
```

Comment:
Maybe this would be more clear as:
```
use_timestamp = conn->txn_global.has_stable_timestamp;
...
if (cval.val != 0 {
    if (use_timestamp) {
    }
}
...
if (use_timestamp) {
}
```
Entirely up to you, feel free to discard without further discussion.

Reply:
We cannot move the `use_timestamp` assignment outside of `if (cval.val != 0)` as that leads to triggering RTS even when the user configured shutdown with `use_timestamp=false`. Currently, I can go with the existing code and later when the `WT_CONN_CLOSING_TIMESTAMP` flag is removed, it can be modified.

Reply:
It seems this change is not necessary? The current code is checking for `WT_CONN_CLOSING_TIMESTAMP`. Your reply to Keith says you can go with the existing code - I think that is better, to minimize changes and also when someone looks at changes they might be confused about why it was changed (when it was not necessary and does not change how it works).

Reply:
Thanks. I reverted the local variable usage and changed it to the existing code.